### PR TITLE
fix(form): safely call validate in document-validity

### PR DESCRIPTION
**Pull request by Yelinz**

### CODE REVIEW

File: packages/form/addon/components/document-validity.js [hunk 1]
```diff
@@ -37,7 +37,9 @@ export default class DocumentValidity extends Component {
   @action
   async validate() {
     await Promise.all(
-      this.args.document.fields.map((field) => field.validate.perform())
+      this.args.document.fields.map(
+        async (field) => await field.validate.perform()
```
Comment (anehx) [inline packages/form/addon/components/document-validity.js:41]: That doesn't do anything.. `field.validate.perform()` already returns a promise..

Comment (Yelinz) [reply packages/form/addon/components/document-validity.js:41]: Getting an error from ember-concurrency that the validate is being canceled as its restartable, form this line of code. Would using enqueue instead be a solution?

Comment (anehx) [reply packages/form/addon/components/document-validity.js:41]: Have you tried using `field.validate.linked().perform()`?

**Review by anehx — APPROVED** [review]
:tada: nice catch!

**Pull request merged**